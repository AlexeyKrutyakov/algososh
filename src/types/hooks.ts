export type useFormProps = {
  name?: string;
  word?: string;
};
