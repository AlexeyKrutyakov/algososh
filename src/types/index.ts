export * from './direction';
export * from './element-states';
export * from './hooks';
export * from './stack';
export * from './queue';
