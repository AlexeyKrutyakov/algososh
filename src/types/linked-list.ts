export type TLinkedList<T> = {
  prepend: (item: T) => void;
  append: (item: T) => void;
  addByIndex: (index: number, item: T) => void;
  deleteByIndex: (index: number) => void;
  deleteHead: () => void;
  deleteTail: () => void;
};
