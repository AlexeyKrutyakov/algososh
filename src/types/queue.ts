export type TQueue<T> = {
  enqueue: (item: T) => void;
  dequeue: () => void;
  getElements: () => T[];
  getSize: () => number;
};
