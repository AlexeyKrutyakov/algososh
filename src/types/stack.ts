export type TStack<T> = {
  push: (item: T) => void;
  pop: () => void;
  clear: () => void;
  getElements: () => T[] | [];
  getSize: () => number;
};
