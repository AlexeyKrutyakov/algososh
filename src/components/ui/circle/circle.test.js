import TestRenderer from 'react-test-renderer';
import { Circle } from './circle';
import { ElementStates } from '../../../types';

describe('Рендеринг компонента Circle', () => {
  it('Circle без буквы отрисовывается без ошибок', () => {
    const tree = TestRenderer.create(<Circle />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Circle с буквами отрисовывается без ошибок', () => {
    const tree = TestRenderer.create(<Circle letter="ABC" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Circle с head отрисовывается без ошибок', () => {
    const tree = TestRenderer.create(
      <Circle
        letter="ABC"
        head="HEAD"
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Circle с react-элементом в head отрисовывается без ошибок', () => {
    const tree = TestRenderer.create(
      <Circle
        letter="ABC"
        head={<Circle letter="R_H" />}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Circle с tail отрисовывается без ошибок', () => {
    const tree = TestRenderer.create(
      <Circle
        letter="ABC"
        tail="TAIL"
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Circle с react-элементом в tail отрисовывается без ошибок', () => {
    const tree = TestRenderer.create(
      <Circle
        letter="ABC"
        tail={<Circle letter="R_T" />}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Circle с index отрисовывается без ошибок', () => {
    const tree = TestRenderer.create(
      <Circle
        letter="ABC"
        index="1"
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Circle с пропом isSmall === true отрисовывается без ошибок', () => {
    const tree = TestRenderer.create(
      <Circle
        letter="ABC"
        isSmall={true}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Circle в состоянии default отрисовывается без ошибок', () => {
    const tree = TestRenderer.create(
      <Circle
        letter="ABC"
        state={ElementStates.Default}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Circle в состоянии changing отрисовывается без ошибок', () => {
    const tree = TestRenderer.create(
      <Circle
        letter="ABC"
        state={ElementStates.Changing}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Circle в состоянии modified отрисовывается без ошибок', () => {
    const tree = TestRenderer.create(
      <Circle
        letter="ABC"
        state={ElementStates.Modified}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
