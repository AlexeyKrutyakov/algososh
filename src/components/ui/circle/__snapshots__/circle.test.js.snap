// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Рендеринг компонента Circle Circle без буквы отрисовывается без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Рендеринг компонента Circle Circle в состоянии changing отрисовывается без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      ABC
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Рендеринг компонента Circle Circle в состоянии default отрисовывается без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      ABC
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Рендеринг компонента Circle Circle в состоянии modified отрисовывается без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      ABC
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Рендеринг компонента Circle Circle с head отрисовывается без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    HEAD
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      ABC
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Рендеринг компонента Circle Circle с index отрисовывается без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      ABC
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    1
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Рендеринг компонента Circle Circle с react-элементом в head отрисовывается без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        >
          R_H
        </p>
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      ABC
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Рендеринг компонента Circle Circle с react-элементом в tail отрисовывается без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      ABC
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        >
          R_T
        </p>
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`Рендеринг компонента Circle Circle с tail отрисовывается без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      ABC
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    TAIL
  </div>
</div>
`;

exports[`Рендеринг компонента Circle Circle с буквами отрисовывается без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      ABC
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Рендеринг компонента Circle Circle с пропом isSmall === true отрисовывается без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      ABC
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
